### PR TITLE
Add fabric capacity monitoring code.

### PR DESCRIPTION
**Pull request by jfeng-arista**
**What I did**
Add fabric capacity monitoring code.

The HLD can be found at 
https://github.com/sonic-net/SONiC/blob/master/doc/voq/fabric.md

The current design does not cover the card hotswap/OIR cases.  We will discuss how to handle that in a separate proposal. 
**Why I did it**

**How I verified it**

**Details if related**

### CODE REVIEW

**Review by arlakshm — COMMENTED** [review]
changes look good. Can we add UT for this?

**Comment by jfeng-arista** [comment]
> changes look good. Can we add UT for this?

added

**Comment by jfeng-arista** [comment]
I checked failures, and they are not Fabric related tests, let me restart the test pipeline

**Comment by jfeng-arista** [comment]
the failed tests not related to fabric

**Comment by jfeng-arista** [comment]
2024-04-25T22:45:55.3666646Z test_mux.py::TestMuxTunnel::test_acl FAILED                           [ 25%]
2024-04-25T22:46:38.3148266Z test_sub_port_intf.py::TestSubPortIntf::test_sub_port_intf_appl_db_proc_seq FAILED                     [ 50%]
2024-04-25T22:47:11.7348687Z test_mux.py::TestMuxTunnel::test_neighbor_miss[standby->ping_serv->active->resolve_entry->delete_entry-IPv6] FAILED                                 [ 75%]
2024-04-25T22:49:30.8390555Z test_vnet.py::TestVnetOrch::test_vnet_orch_2 FAILED                           [ 25%]
2024-04-25T22:50:12.9072929Z test_twamp.py::TestTwampLight::test_SenderContinuousMulti FAILED                     [ 50%]
2024-04-25T22:50:46.0160473Z test_srv6.py::TestSrv6::test_srv6 FAILED                                 [ 75%]
2024-04-25T22:51:39.4822773Z test_srv6.py::TestSrv6Mysid::test_mysid FAILED [ 80%]
2024-04-25T22:52:40.1379025Z test_srv6.py::TestSrv6::test_srv6 FAILED [ 90%]

above are the failed tests, looks like not related to this change

File: orchagent/fabricportsorch.cpp [hunk 1]
```diff
@@ -836,6 +837,185 @@ void FabricPortsOrch::updateFabricDebugCounters()
...
+            linkIssue = true;
+        }
+
+        if (lnkStatus == "down" || linkIssue == true)
+        {
+            downCapacity += 44316;
+        }
+        else
+        {
+            capacity += 44316;
```
Comment (arlakshm) [inline orchagent/fabricportsorch.cpp:932]: where is  is `44316` coming from?

Comment (jfeng-arista) [reply orchagent/fabricportsorch.cpp:932]: that's the  90% of the enocdded serdes rate .

File: orchagent/fabricportsorch.cpp [hunk 2]
```diff
@@ -836,6 +837,185 @@ void FabricPortsOrch::updateFabricDebugCounters()
...
+       // total fabric capacity.
+        bool linkIssue = false;
+        if (configIsolated == "1" || isolated == "1" || autoIsolated == "1")
+        {
+            linkIssue = true;
+        }
+
+        if (lnkStatus == "down" || linkIssue == true)
+        {
+            downCapacity += 44316;
```
Comment (prsunny) [inline orchagent/fabricportsorch.cpp:928]: Do not recommend hardcoding magic numbers in code. Please define this.

Comment (jfeng-arista) [reply orchagent/fabricportsorch.cpp]: > Do not recommend hardcoding magic numbers in code. Please define this.

done

**Review by prsunny — APPROVED** [review]

**Review by arlakshm — APPROVED** [review]

**Pull request merged**